File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int begin=0,end=nums.size()-1;
        while(begin<=end){
            int mid = (begin+end)/2;
            
            if(nums[mid]==target) return mid;
            else if(target>nums[mid]){
                begin=mid+1;
            }
            else{
                end=mid-1;
            }
        }
        return -1;
    }
};

File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        return fun(nums,target,0,nums.length-1);
    }
    public int fun(int nums[],int target,int begin,int end){
        int mid = (begin+end)/2;
        if(begin>end) return -1;
        if(nums[mid] == target) return mid;
        else if(target>nums[mid]) return fun(nums,target,mid+1,end);
        else return fun(nums,target,begin,mid-1);
        // return -1;
    }
}